frontend/src/Matching/CodeSpace.tsx

```diff
@@ -19,27 +23,57 @@ const CodeSpace = () => {
   const { socketId, difficulty } = location.state || {};
   const [socket, setSocket] = useState<Socket | null>(null);
   const [question, setQuestion] = useState<Question | null>(null);
+  const [code, setCode] = useState("console.log('hello world!');");
+  const [value, setValue] = React.useState("console.log('hello world!');");
+  const [selectedLanguage, setSelectedLanguage] = useState('c'); // Default language is C
+
+  const onChange = React.useCallback((val: string, viewUpdate: any) => {
+    console.log('val:', val);
+    setValue(val);
+
```

Comment:
(2/2) Suggestion to use the latest code during refresh. 
Previous code will always show `console.log('hello world!');` on refresh.

```suggestion
// Save the code value to localStorage
    localStorage.setItem('code', val);
```